File: snippets/fsharp/System/String/IsNullOrEmpty/inoe.fs
namespace IsNullOrEmpty
open System

module Inoe =

    //<Snippet1>
    let test (s: string): string =
        if String.IsNullOrEmpty(s)
        then "is null or empty"
        else $"(\"{s}\") is neither null nor empty"

    let s1 = "abcd"
    let s2 = ""
    let s3 = null

    printfn "String s1 %s" (test s1)
    printfn "String s2 %s" (test s2)
    printfn "String s2 %s" (test s3)

    // The example displays the following output:
    //       String s1 ("abcd") is neither null nor empty.
    //       String s2 is null or empty.
    //       String s3 is null or empty.
    //</Snippet1>

File: snippets/csharp/System/String/IsNullOrEmpty/inoe.cs
using System;

class Sample
{
    public static void Main()
    {
        //<snippet1>
        string s1 = "abcd";
        string s2 = "";
        string s3 = null;

        Console.WriteLine($"String s1 {Test(s1)}.");
        Console.WriteLine($"String s2 {Test(s2)}.");
        Console.WriteLine($"String s3 {Test(s3)}.");

        string Test(string s)
        {
            if (string.IsNullOrEmpty(s))
                return "is null or empty";
            else
                return $"(\"{s}\") is neither null nor empty";
        }

        // The example displays the following output:
        //       String s1 ("abcd") is neither null nor empty.
        //       String s2 is null or empty.
        //       String s3 is null or empty.
        // </snippet1>
    }
}
